test/envoye2e/stackdriver_plugin/stackdriver_test.go

```diff
@@ -895,3 +895,65 @@ func TestStackdriverRbacAccessDenied(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestStackdriverMetricExpiry(t *testing.T) {
+	params := driver.NewTestParams(t, map[string]string{
+		"ServiceAuthenticationPolicy": "NONE",
+		"StackdriverRootCAFile":       driver.TestPath("testdata/certs/stackdriver.pem"),
+		"StackdriverTokenFile":        driver.TestPath("testdata/certs/access-token"),
+	}, envoye2e.ProxyE2ETests)
+
+	sdPort := params.Ports.Max + 1
+	stsPort := params.Ports.Max + 2
+	params.Vars["SDPort"] = strconv.Itoa(int(sdPort))
+	params.Vars["STSPort"] = strconv.Itoa(int(stsPort))
+	params.Vars["ClientMetadata"] = params.LoadTestData("testdata/client_node_metadata.json.tmpl")
+	params.Vars["ServerMetadata"] = params.LoadTestData("testdata/server_node_metadata.json.tmpl")
+	params.Vars["ClientHTTPFilters"] = driver.LoadTestData("testdata/filters/mx_outbound.yaml.tmpl")
+	params.Vars["ServerHTTPFilters"] = driver.LoadTestData("testdata/filters/mx_inbound.yaml.tmpl") +
+		driver.LoadTestData("testdata/filters/stackdriver_inbound.yaml.tmpl")
+	sd := &Stackdriver{Port: sdPort}
+	if err := (&driver.Scenario{
+		Steps: []driver.Step{
+			&driver.XDS{},
+			sd,
+			&SecureTokenService{Port: stsPort},
+			&driver.Update{Node: "client", Version: "0", Listeners: []string{
+				params.LoadTestData("testdata/listener/client.yaml.tmpl"),
+			}},
+			&driver.Update{Node: "server", Version: "0", Listeners: []string{
+				params.LoadTestData("testdata/listener/server.yaml.tmpl"),
+			}},
+			&driver.Envoy{Bootstrap: params.LoadTestData("testdata/bootstrap/server.yaml.tmpl")},
+			&driver.Envoy{Bootstrap: params.LoadTestData("testdata/bootstrap/client.yaml.tmpl")},
+			&driver.Sleep{Duration: 1 * time.Second},
+			&driver.Repeat{N: 10,
+				Step: &driver.HTTPCall{
+					Port: params.Ports.ClientPort,
+					Body: "hello, world!",
+				},
+			},
+			sd.Check(params,
+				[]string{"testdata/stackdriver/server_request_count.yaml.tmpl"},
+				[]SDLogEntry{}, nil, true,
+			),
+			sd.Reset(),
+			&driver.Sleep{Duration: 10 * time.Second},
```

Comment:
any way to speed this up? 10s seems like a lot.

Reply:
Purge happens lazily on report cycle, and the report cycle time is set to 20s in order to accommodate slow asan/tsan build: https://github.com/istio/proxy/pull/2806/commits/9fda46f0d12df049f6c27bf2d64278dda3d7ba06

Reply:
Yeah, SD test takes about a minute so it's not making it worse. It can be run in parallel if necessary.